activerecord/lib/active_record/type/serialized.rb

```diff
@@ -2,7 +2,7 @@
 
 module ActiveRecord
   module Type
-    class Serialized < DelegateClass(ActiveModel::Type::Value) # :nodoc:
+    class Serialized < SimpleDelegator # :nodoc:
```

Comment:
This means that the base class for serialized types won't be an `ActiveModel::Type::Value` anymore. I can't think of a problem with doing this but maybe someone more familiar with AR internals can weigh in here (maybe @fatkodima or @eileencodes?)

Reply:
Do we need to make this change?

Reply:
I made those changes and all tests still pass:

```diff
diff --git a/activerecord/lib/active_record/encryption/encryptable_record.rb b/activerecord/lib/active_record/encryption/encryptable_record.rb
index 01328ec663..5c8dfe3499 100644
--- a/activerecord/lib/active_record/encryption/encryptable_record.rb
+++ b/activerecord/lib/active_record/encryption/encryptable_record.rb
@@ -88,14 +88,14 @@ def encrypt_attribute(name, key_provider: nil, key: nil, deterministic: false, s
               scheme = scheme_for key_provider: key_provider, key: key, deterministic: deterministic, support_unencrypted_data: support_unencrypted_data, \
                 downcase: downcase, ignore_case: ignore_case, previous: previous, compress: compress, compressor: compressor, **context_properties
 
+              type_options = { scheme: scheme, default: columns_hash[name.to_s]&.default }
+
               if cast_type.serialized?
-                cast_type.tap do |serialized_type|
-                  serialized_type.replace_serialized_subtype do |current_subtype|
-                    ActiveRecord::Encryption::EncryptedAttributeType.new(scheme: scheme, cast_type: current_subtype, default: columns_hash[name.to_s]&.default)
-                  end
+                cast_type.replace_serialized_subtype do |current_subtype|
+                  ActiveRecord::Encryption::EncryptedAttributeType.new(cast_type: current_subtype, **type_options)
                 end
               else
-                ActiveRecord::Encryption::EncryptedAttributeType.new(scheme: scheme, cast_type: cast_type, default: columns_hash[name.to_s]&.default)
+                ActiveRecord::Encryption::EncryptedAttributeType.new(cast_type: cast_type, **type_options)
               end
             end
 
diff --git a/activerecord/lib/active_record/type/serialized.rb b/activerecord/lib/active_record/type/serialized.rb
index b1d9a79266..b5a359149c 100644
--- a/activerecord/lib/active_record/type/serialized.rb
+++ b/activerecord/lib/active_record/type/serialized.rb
@@ -2,7 +2,7 @@
 
 module ActiveRecord
   module Type
-    class Serialized < SimpleDelegator # :nodoc:
+    class Serialized < DelegateClass(ActiveModel::Type::Value) # :nodoc:
       undef to_yaml if method_defined?(:to_yaml)
 
       include ActiveModel::Type::Helpers::Mutable
@@ -12,7 +12,7 @@ class Serialized < SimpleDelegator # :nodoc:
       def initialize(subtype, coder)
         @subtype = subtype
         @coder = coder
-        __setobj__(subtype)
+        super(subtype)
       end
 
       def init_with(coder) # :nodoc:
@@ -67,6 +67,7 @@ def serialized? # :nodoc:
       def replace_serialized_subtype(&block) # :nodoc:
         @subtype = block.call(subtype)
         __setobj__(@subtype)
+        self
       end
 
       private
```

Reply:
Ah yes that works much better, if you could apply that change 👍

With that, I think this seems like the best approach to fixing this. 

I've updated the changelog and encryption guides and I'll close the other PR.

Reply:
@rafaelfranca - I've applied your suggestion here - https://github.com/rails/rails/pull/52650/commits/91745df8997750e77873e7ebd48722023ed4e409 🙏